File: TicketOn.Server/Entidades/Reventa.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations.Schema;

namespace TicketOn.Server.Entidades
{
    public class Reventa
    {
        public int Id { get; set; }  // ID de la reventa
        public string Estado {  get; set; } = "Disponible";
        public int EntradaVentaId { get; set; }  // ID de la entrada venta
        public EntradaVenta EntradaVenta { get; set; }

        public string UsuarioId { get; set; }  // ID del usuario que está revendiendo
        public IdentityUser Usuario { get; set; }  // Relación con el usuario
        public string? CompradorId { get; set; }
        public IdentityUser Comprador { get; set; }

        public decimal PrecioReventa { get; set; }  // Precio de reventa
        public DateTime FechaPublicacion { get; set; } = DateTime.UtcNow;  // Fecha de publicación de la reventa
        public DateTime? FechaReventa { get; set; }
    }
}

File: ticketon.client/src/app/reventa/reventa.ts
export interface Reventa {
  id: number;
  entradaVentaId: number;
  precioReventa: number;
  usuarioId: string;
  estado: string; // Estado de la reventa (Disponible, Vendida, Cancelada)
  compradorId?: string; // ID del comprador si la reventa fue completada
  fechaPublicacion?: Date;
  fechaReventa?: Date; // Fecha en la que se completó la reventa (opcional)
}

export interface ReventaCreacionDTO {
  entradaVentaId: number; // ID de la entrada a revender
  precioReventa: number;   // Precio por el que se revende
  usuarioId?: string;       // ID del usuario que está revendiendo
}

export interface ReventaDTO {
  id: number;
  entradaVentaId: number;
  precioReventa: number;
  usuarioId: string;
  estado: string;
  fechaPublicacion: Date;
  nombreEvento: string;  // Nuevo campo
  imagenEvento: string; // Fecha de publicación de la reventa
}

